modules/dkan/dkan_dataset/modules/dkan_dataset_rest_api/dkan_dataset_rest_api.module

```diff
@@ -35,3 +35,184 @@ function dkan_dataset_rest_api_services_request_postprocess_alter($controller, $
     }
   }
 }
+
+/**
+ * Implements hook_services_resources_alter().
+ */
+function dkan_dataset_rest_api_services_resources_alter(&$resources, $endpoint) {
+  if (isset($resources['node']['operations']['create'])) {
+    $resources['node']['operations']['create']['callback'] =
+      "_dkan_dataset_rest_api_resource_create";
+  }
+
+  if (isset($resources['node']['operations']['update'])) {
+    $resources['node']['operations']['update']['callback'] =
+      "_dkan_dataset_rest_api_resource_update";
+  }
+}
+
+/**
+ * Services operation callback: node creation.
+ */
+function _dkan_dataset_rest_api_resource_create($request) {
+  $node_array = _services_arg_value($request, 'node');
+  $node = (object) $node_array;
+
+  if (isset($node->type)) {
+    if ($node->type == "dataset" || $node->type == "resource") {
+
+      try {
+        field_attach_validate('node', $node);
+        _dkan_dataset_rest_api_field_validation($node);
+      }
+      catch (\Exception $e) {
+        return services_error($e->getMessage());
+      }
+
+      node_save($node);
+
+      $nid = $node->nid;
+      $response = array('nid' => $nid);
+      if ($uri = services_resource_uri(array('node', $nid))) {
+        $response['uri'] = $uri;
+      }
+    }
+    else {
+      $response = _node_resource_create($request);
+    }
+  }
+  else {
+    return services_error("Type not set for node");
+  }
+
+  return $response;
+}
+
+/**
+ * Services operation callback: node update.
+ */
+function _dkan_dataset_rest_api_resource_update($nid, $request) {
+  $node = node_load($nid);
+
+  if (isset($node->type)) {
+    if ($node->type == "dataset" || $node->type == "resource") {
+
+      if (empty($node->nid)) {
+        return services_error(t('Node @nid not found', array('@nid' => $node->nid)), 404);
```

Comment:
Seems like a strange error to throw if the nid is empty - it's just always going to read "Node  not found"